src/main/java/io/confluent/connect/elasticsearch/RetryUtil.java

```diff
@@ -136,39 +144,43 @@ public static long computeRetryWaitTimeInMillis(int retryAttempts,
    * @param <T>              the return type of the function to retry
    * @return the function's return value
    * @throws ConnectException        if the function failed after retries
-   * @throws Exception               if the function throws an exception
    */
   protected static <T> T callWithRetries(
       String description,
       Callable<T> function,
       int maxAttempts,
       long initialBackoff,
+      long connectionTimeout,
       Time clock
   ) throws Exception {
     assert description != null;
     assert function != null;
-
+    log.warn("in call with retries");
     int attempt = 0;
+    long backoff = computeRandomRetryWaitTimeInMillis(attempt, initialBackoff);
     while (true) {
       ++attempt;
+      ExecutorService executorService = Executors.newSingleThreadExecutor();
       try {
-        log.trace(
+        log.warn(
             "Try {} (attempt {} of {})",
             description,
             attempt,
             maxAttempts
         );
-        return function.call();
-      } catch (IOException e) {
+        Future<T> future = executorService.submit(function);
+        // Wait for connectionTimeout duration until the result is returned
+        return future.get(connectionTimeout, TimeUnit.MILLISECONDS);
```

Comment:
@ilanjiR  its a right find that the bulk isn't setting timeout but its a fix in the wrong place.  We need to instead find and fix why we are passing RequestOptions.DEFAULT. instead of a real RequestOptions with timeout and we need to fix this in other places also that are doing the same.  CC: @erdody

```
BulkResponse bulkResponse = callWithRetries(
    "retrying bulk request",
    () -> client.bulk(request, RequestOptions.DEFAULT)
);
```

Reply:
It seems as per docs RequestOptions here are to override the timeouts defined when creating the client and passing DEFAULT means no override, in that case, we should find out why the timeouts defined when creating the client aren't working here.